internal/adapter/middleware/auth_middleware.go

```diff
@@ -31,5 +31,6 @@ func AuthMiddleware(authService port.AuthService) gin.HandlerFunc {
 
 		ctx.Set("userDNI", claims["sub"])
 		ctx.Set("userRole", claims["role"])
+		ctx.Set("adminPermission", false)
```

Comment:
We do not need `AdminMiddleware` to know if the role is Admin or not. Please take a look at line 33, we have userRole, we can simply validate if this value has the value of Admin Role.